DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -308,6 +314,90 @@ private DocumentsRunner.CompletionStatus migrateDocumentsWithOneWorker(SourceRep
+            CreateSnapshot.run(
+                    c -> new FileSystemSnapshotCreator(SNAPSHOT_NAME, c, SearchClusterContainer.CLUSTER_SNAPSHOT_DIR),
+                    new OpenSearchClient(esSourceContainer.getUrl(), null),
+                    false);
+            var tempDirSnapshot = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_snapshot");
+            var tempDirLucene = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_lucene");
+
+            String[] args = {
+                "--snapshot-name", SNAPSHOT_NAME,
+                "--snapshot-local-dir", tempDirSnapshot.toString(),
+                "--lucene-dir", tempDirLucene.toString(),
+                "--target-host", osTargetContainer.getHttpHostAddress()
+            };
+
+            try {
+                esSourceContainer.copySnapshotData(tempDirSnapshot.toString());
+
+                var targetClient = new OpenSearchClient(osTargetContainer.getHttpHostAddress(), null);
+                var sourceRepo = new FileSystemRepo(tempDirSnapshot);
+                migrateMetadata(sourceRepo, targetClient, SNAPSHOT_NAME, INDEX_ALLOWLIST);
+
+                String classpath = System.getProperty("java.class.path");
+                String javaHome = System.getProperty("java.home");
+                String javaExecutable = javaHome + File.separator + "bin" + File.separator + "java";
+
+                // Kick off the doc migration process
+                log.atInfo().setMessage("Running RfsMigrateDocuments with args: " + Arrays.toString(args)).log();
+                ProcessBuilder processBuilder = new ProcessBuilder(
+                        javaExecutable, "-cp", classpath, "com.rfs.RfsMigrateDocuments"
+                );
+                processBuilder.command().addAll(Arrays.asList(args));
+
+                Process process = processBuilder.start();
+                log.atInfo().setMessage("Process started with ID: " + Long.toString(process.toHandle().pid())).log();
+                
+                // Kill the process and fail if we have to wait too long
+                boolean finished = process.waitFor(2, TimeUnit.MINUTES);
+                if (!finished) {
+                    log.atError().setMessage("Process timed out, attempting to kill it...").log();
+                    process.destroy(); // Try to be nice about things first...
+                    if (!process.waitFor(10, TimeUnit.SECONDS)) {
+                        log.atError().setMessage("Process still running, attempting to force kill it...").log();
+                        process.destroyForcibly(); // ..then avada kedavra
+                    }
+                    Assertions.fail("The process did not finish within the timeout period.");
+                }
+
+                int exitCode = process.exitValue();
+
+                // Check if the exit code is as expected
+                int expectedExitCode = 0;
```

Comment:
this would need to be parameterized for the error case.

Reply:
Yep!